### PR TITLE
refactor(lint): add flake8-raises

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

Enables RSE102, TRY201, and B904 in ruff which correspond to the three errors that flake8-raises provides.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/ext/commands/common_bot_base.py [hunk 1]
```diff
@@ -463,7 +463,7 @@ def _resolve_name(self, name: str, package: Optional[str]) -> str:
         try:
             return importlib.util.resolve_name(name, package)
         except ImportError:
-            raise errors.ExtensionNotFound(name)
+            raise errors.ExtensionNotFound(name) from None
```
Comment (shiftinv) [inline disnake/ext/commands/common_bot_base.py:466]: this should keep the original error

File: disnake/ext/commands/converter.py [hunk 2]
```diff
@@ -431,9 +431,9 @@ async def convert(self, ctx: AnyContext, argument: str) -> disnake.Message:
         try:
             return await channel.fetch_message(message_id)
         except disnake.NotFound:
-            raise MessageNotFound(argument)
+            raise MessageNotFound(argument) from None
         except disnake.Forbidden:
-            raise ChannelNotReadable(channel)  # type: ignore
+            raise ChannelNotReadable(channel) from None  # type: ignore
```
Comment (shiftinv) [inline disnake/ext/commands/converter.py:436]: Maybe consider the original error here as well? Not sure.

File: disnake/ext/commands/common_bot_base.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/ext/commands/common_bot_base.py]: https://github.com/DisnakeDev/disnake/pull/950/commits/a31297b5838c1746d1d663b4a7c907eb2437d7f1

File: disnake/ext/commands/converter.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/ext/commands/converter.py:436]: I don't think its worth keeping as these are just reskinned 403s and 404s

**Review by shiftinv — APPROVED** [review]

**Review by shiftinv — CHANGES_REQUESTED** [review]
Since this changes runtime behavior, a short changelog entry would be great

**Review by shiftinv — APPROVED** [review]

**Pull request merged**